src/features/room/sagas/getPrivacy.saga.ts

```diff
@@ -1,31 +1,40 @@
 import { put } from 'redux-saga/effects'
-import { Stream, StreamPermission } from 'streamr-client'
+import { StreamPermission } from 'streamr-client'
 import { RoomAction } from '..'
-import { PrivacySetting } from '$/types'
+import { EnhancedStream, PrivacySetting } from '$/types'
 import RoomNotFoundError from '$/errors/RoomNotFoundError'
 import getStream from '$/utils/getStream'
 import handleError from '$/utils/handleError'
 import takeEveryUnique from '$/utils/takeEveryUnique'
+import { isTokenGatedRoom } from '$/features/tokenGatedRooms/utils/isTokenGatedRoom'
 
 function* onGetPrivacyAction({
     payload: { roomId, streamrClient },
 }: ReturnType<typeof RoomAction.getPrivacy>) {
     try {
-        const stream: undefined | Stream = yield getStream(streamrClient, roomId)
+        const stream: undefined | EnhancedStream = yield getStream(streamrClient, roomId)
 
         if (!stream) {
             throw new RoomNotFoundError(roomId)
         }
 
+        const isTokenGated: boolean = yield isTokenGatedRoom(roomId, streamrClient)
```

Comment:
Can we have an alternative (synchronous maybe) implementation of this check to skip the additional fetching? We already hold a copy of \*the\* `stream`.